src/nbl/asset/ICPUDescriptorSet.cpp

```diff
@@ -0,0 +1,253 @@
+				continue;
+
+			auto descriptorInfos = m_descriptorInfos[t]->begin();
+			assert(descriptorInfos);
+
+			const auto category = getCategoryFromType(type);
+			for (uint32_t i = 0u; i < descriptorCount; ++i)
+			{
+				switch (category)
+				{
+				case IDescriptor::E_CATEGORY::EC_BUFFER:
+					static_cast<asset::ICPUBuffer*>(descriptorInfos[i].desc.get())->convertToDummyObject(referenceLevelsBelowToConvert);
+					break;
+
+				case IDescriptor::E_CATEGORY::EC_IMAGE:
+				{
+					static_cast<asset::ICPUImageView*>(descriptorInfos[i].desc.get())->convertToDummyObject(referenceLevelsBelowToConvert);
+					if (descriptorInfos[i].info.image.sampler)
+						descriptorInfos[i].info.image.sampler->convertToDummyObject(referenceLevelsBelowToConvert);
+				} break;
+
+				case IDescriptor::EC_BUFFER_VIEW:
+					static_cast<asset::ICPUBufferView*>(descriptorInfos[i].desc.get())->convertToDummyObject(referenceLevelsBelowToConvert);
+					break;
+
+				default:
+					assert(!"Invalid code path.");
+				}
+			}
+		}
+	}
+}
+
+void ICPUDescriptorSet::restoreFromDummy_impl(IAsset* _other, uint32_t _levelsBelow)
+{
+	auto* other = static_cast<ICPUDescriptorSet*>(_other);
+
+	if (_levelsBelow)
+	{
+		--_levelsBelow;
+		restoreFromDummy_impl_call(m_layout.get(), other->getLayout(), _levelsBelow);
+
+		for (uint32_t t = 0u; t < static_cast<uint32_t>(IDescriptor::E_TYPE::ET_COUNT); ++t)
+		{
+			const auto type = static_cast<IDescriptor::E_TYPE>(t);
+			const auto descriptorCount = m_layout->getTotalDescriptorCount(type);
+			if (descriptorCount == 0ull)
+				continue;
+
+			auto descriptorInfos = m_descriptorInfos[t]->begin();
+			assert(descriptorInfos);
+
+			auto otherDescriptorInfos = other->m_descriptorInfos[t]->begin();
+
+			const auto category = getCategoryFromType(type);
+			for (uint32_t i = 0u; i < descriptorCount; ++i)
+			{
+				switch (category)
+				{
+				case IDescriptor::E_CATEGORY::EC_BUFFER:
+					restoreFromDummy_impl_call(static_cast<ICPUBuffer*>(descriptorInfos[i].desc.get()), static_cast<ICPUBuffer*>(otherDescriptorInfos[i].desc.get()), _levelsBelow);
+					break;
+
+				case IDescriptor::EC_IMAGE:
+				{
+					restoreFromDummy_impl_call(static_cast<ICPUImageView*>(descriptorInfos[i].desc.get()), static_cast<ICPUImageView*>(otherDescriptorInfos[i].desc.get()), _levelsBelow);
+					if (descriptorInfos[i].info.image.sampler && otherDescriptorInfos[i].info.image.sampler)
+						restoreFromDummy_impl_call(descriptorInfos[i].info.image.sampler.get(), otherDescriptorInfos[i].info.image.sampler.get(), _levelsBelow);
+				} break;
+
+				case IDescriptor::EC_BUFFER_VIEW:
+					restoreFromDummy_impl_call(static_cast<ICPUBufferView*>(descriptorInfos[i].desc.get()), static_cast<ICPUBufferView*>(otherDescriptorInfos[i].desc.get()), _levelsBelow);
+					break;
+
+				default:
+					assert(!"Invalid code path.");
+				}
+			}
+		}
+	}
+}
+
+bool ICPUDescriptorSet::isAnyDependencyDummy_impl(uint32_t _levelsBelow) const
+{
+	--_levelsBelow;
```

Comment:
missing `_levelsBelow` check before proceeding

please make it
```cpp
--_levelsBelow;
if (!_levelsBelow)
    return true;
```

Reply:
Resolved.